### PR TITLE
Replace jQuery tab visibility with Vue reactivity (Fixes #4700)

### LINKED ISSUE #4700: Replace tab visibility jQuery with Vue reactivity for expert mode and feature-controlled tabs
### Describe the bug

The tab bar currently relies on jQuery to show/hide tabs (e.g., expert-only or feature-dependent tabs). This duplicates logic that can be handled more robustly by Vue. The mixed jQuery/Vue state can cause:

- UI desync when reactive state changes but DOM is toggled by jQuery.
- Hard-to-maintain logic duplicated across files.
- Flicker or wrong visibility on reconnect or after content reloads.

### To Reproduce

- Launch the Configurator and connect to an FC.
- Toggle the Expert Mode checkbox and observe tab visibility.
- Change FC build options (e.g., GPS, OSD) or connect boards with different features.
- Navigate between tabs or reconnect.

### Expected behavior

- Tab visibility should respond to reactive Vue state (expertMode, FC.CONFIG.buildOptions, FC.FEATURE_CONFIG.features) without jQuery manipulating DOM visibility.

- Expert-only and feature-only tabs are conditionally rendered via Vue (v-show) consistently on connect/disconnect and tab switches.

### Configurator version

2025.12.0-beta

### Flight controller configuration

```txt

```

### Add any other context about the problem that you think might be relevant here

_No response_

### PR DESCRIPTION
**Pull request by Erics1337**
# Replace jQuery tab toggles with Vue reactivity (Fixes #4700)

## Overview
Moves tab visibility logic from jQuery to Vue so Expert Mode and feature-based tabs update instantly and stay in sync without DOM flicker.

## What changed
- Added `expertMode` to the Vue model and bound the checkbox with `v-model`.
- Used `v-show` for visibility:
  - Expert-only: failsafe, adjustments, sensors, logging
  - Build options: servos (USE_SERVOS/USE_WING), gps (USE_GPS)
  - Features: osd, led_strip, transponder via `features.isEnabled(...)`
- Removed jQuery `.toggle()` calls in [updateTabList.js](cci:7://file:///Users/ericswanson/code/open-source/betaflight-configurator/src/js/utils/updateTabList.js:0:0-0:0); it now just syncs the checkbox to Vue state.

## Why
- One source of truth (Vue) instead of mixing jQuery and Vue.
- Fewer edge cases during connect/disconnect and tab switching.
- Simpler and more maintainable—the rules live next to the markup.

## Test
- Toggle Expert Mode: tabs appear/disappear immediately.
- Connect different FCs: gps/servos/osd/led_strip/transponder tabs reflect build options and features.

Fixes #4700

## Summary by CodeRabbit

* **New Features**
  * Introduced expert mode toggle to reveal advanced configuration options

* **Changes**
  * Tab visibility reorganized: failsafe, sensors, and logging tabs now visible only in expert mode
  * Availability of adjustments, servos, GPS, OSD, transponder, and LED strip tabs depends on firmware build configuration and enabled features

### CODE REVIEW

**Review by VitroidFPV — APPROVED** [review]
This is something that I've been playing with locally, along with loading the tabs themselves with Vue directly. I don't have much time to work on the more complex behind-the-scenes stuff nowadays, I'm thankful that there's effort going into this still :+1:

**Comment by VitroidFPV** [comment]
Going forward this would be a great way to gate the tab content itself, but this is a great step forward already

File: src/js/utils/updateTabList.js [hunk 1]
```diff
@@ -1,19 +1,14 @@
...
+export function updateTabList(_features) {
+    try {
+        const isExpertModeEnabled = $('input[name="expertModeCheckbox"]').is(":checked");
+        if (window.vm && typeof window.vm.expertMode !== "undefined") {
+            window.vm.expertMode = isExpertModeEnabled;
+        }
+    } catch {
+        // noop: if Vue model not available, do nothing
+    }
 }
```
[the remark this answers is not in the document]
Comment (haslinghuis) [reply src/js/utils/updateTabList.js:14]: @coderabbitai provide a actionable commit with these changes
